pkg/provider/cloud/vsphere/provider.go

```diff
@@ -181,14 +181,20 @@ func (v *VSphere) ValidateCloudSpec(ctx context.Context, spec kubermaticv1.Cloud
 	}
 	defer session.Logout(ctx)
 
-	effectiveDatastore := v.dc.DefaultDatastore
```

Comment:
But wouldn't make sense to simply remove this property from the seed object ? then this code will work just fine. I find it a bit confusing, to let the code address incorrect configurations. If you don't have permissions to access the default datastore then simply remove it.

Reply:
The `DefaultDatastore` is mandatory field in the dc spec, besides that what if we have a scenario where we want the default datastore for one user cluster but datastore cluster in another user cluster in the same dc.
Currently if datastore cluster is specified in the cluster spec even then the validation is done to check access to the default datastore but admin doesn't want the user to have access to the default datastore and only have access to the datastore cluster specified in the cluster spec.

Reply:
This is quite strange. I don't remember that this property is required. Alright, I mean in general I also don't know how the CSI driver would behave in terms of using the datastore cluster. Have you tried to use only the datastore cluster and created volumes ?

Reply:
I tried creating cluster with datastore cluster but nodes didn't come up as the datastore cluster we have currently isn't functional but KKP didn't try to validate access to default datastore of the DC when the datastore cluster was specified for the cluster & the cluster creation worked which is the what we are trying to achive with this PR that if the datastore cluster is specified in cluster spec then we don't validate access to default datastore as this blocks cluster creation if the user doesn't have access to default datastore.

Reply:
Did we merge this by accident when I removed the code freeze label?

Reply:
yes, it had an lgtm/approved so got merged after the code freeze label was removed.